packages/kit/src/utils/import.js

```diff
@@ -3,17 +3,22 @@ import { pathToFileURL } from 'node:url';
 
 /**
  * Resolve a dependency relative to the current working directory,
- * rather than relative to this package
+ * rather than relative to this package (but falls back to trying that, if necessary)
  * @param {string} dependency
  */
-export function resolve_peer_dependency(dependency) {
+export async function resolve_peer_dependency(dependency) {
 	try {
 		// @ts-expect-error the types are wrong
 		const resolved = imr.resolve(dependency, pathToFileURL(process.cwd() + '/dummy.js'));
-		return import(resolved);
+		return await import(resolved);
```

Comment:
By the way

How about focusing logic on the try syntax?
```suggestion
		return await import(resolved).catch(() => import(dependency));
```